### PR TITLE
vlans: refactored and bug free

### LINKED ISSUE #475: minimega: crash in vlan code
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0x461f23]

goroutine 64132 [running]:
runtime.panic(0xa85960, 0x143b4c8)
    /usr/lib/go/src/pkg/runtime/panic.c:266 +0xb6
main.(*AllocatedVLANs).allocate(0xc2102091e0, 0xc21203f6c0, 0xc, 0xc)
    /opt/minimega/src/minimega/vlans.go:101 +0x293
main.(*AllocatedVLANs).ParseVLAN(0xc2102091e0, 0xc21203f6c0, 0xc, 0x1, 0x0, ...)
    /opt/minimega/src/minimega/vlans.go:343 +0x5e8
main.processVMNet(0xc211cee790, 0xf, 0x0, 0x0, 0x0, ...)
    /opt/minimega/src/minimega/misc.go:410 +0x411
main.func·023(0xab22a0, 0x14558a0, 0xc212f38f60, 0x3, 0xc21000b490)
    /opt/minimega/src/minimega/vmconfig.go:90 +0x131
main.cliVmConfigField(0xc212f38f60, 0xb14480, 0x3, 0x1463120)
    /opt/minimega/src/minimega/vmconfig_cli.go:557 +0x1aa
main.func·058(0xc212f38f60, 0x4a353f)
    /opt/minimega/src/minimega/vmconfig_cli.go:135 +0x3b
main.func·084(0xc212f38f60, 0xc212f3c180)
    /opt/minimega/src/minimega/cli.go:95 +0x2e
minicli.func·010()
    /opt/minimega/src/minicli/minicli.go:130 +0x58
created by minicli.ProcessCommand
    /opt/minimega/src/minicli/minicli.go:139 +0x141

```

### PR DESCRIPTION
**Pull request by jcrussell**
Split the vlan code out into a separate package. Found and fixed the bug that was causing the crash in #475.

Passes `minitest` and `go test`.

Closes #475.

@glattercj @djfritz

### CODE REVIEW

File: src/minimega/bridge.go [hunk 1]
```diff
@@ -25,10 +25,7 @@ import (
 const (
 	DisconnectedVLAN = -1
 	TrunkVLAN        = -2
-)
-
-const (
-	TAP_REAP_RATE = time.Second
+	TAP_REAP_RATE    = time.Second
```
Comment (djfritz) [inline src/minimega/bridge.go]: This is fine but I tend to split groups of constants into their logical groupings. The stdlib does this too (groups of things that are IOTA'd, etc.)

File: src/minimega/misc.go [hunk 2]
```diff
@@ -407,7 +393,7 @@ func processVMNet(spec string) (res NetConfig, err error) {
 
 	log.Debug("vm_net got b=%v, v=%v, m=%v, d=%v", b, v, m, d)
```
Comment (djfritz) [inline src/minimega/misc.go]: vm_net is a very old reference...

**Comment by djfritz** [comment]
Holy cow. 

Does this need more testing before merging? It otherwise LGTM

**Comment by jcrussell** [comment]
I've done some local and CCC tests. Seems to be working fine.

**Pull request merged**